File: Runtime选择器&方法/Runtime选择器&方法/ViewController.m
//
//  ViewController.m
//  Runtime选择器&方法
//
//  Created by 陈胜华 on 16/8/15.
//  Copyright © 2016年 oneyd.me. All rights reserved.
//

#import "ViewController.h"


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    SEL sel = @selector(method);
    
    NSLog(@"sel : %p",sel);
    
}

- (void)method {
    
}

@end

File: Blocks_NetWork_Swift/Blocks_NetWork_Swift/ViewController.swift
//
//  ViewController.swift
//  Blocks_NetWork_Swift
//
//  Created by 陈胜华 on 16/12/1.
//  Copyright © 2016年 oneyd.me. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.red
        
        self.navigationItem.rightBarButtonItem = UIBarButtonItem.init(title: "Push", style: .plain, target: self, action: #selector(ViewController.pushAction))
        self.navigationItem.leftBarButtonItem = UIBarButtonItem.init(title: "循环引用", style: .plain, target: self, action: #selector(pushLoopVC))
    }
    
    final func pushAction() {
        let blockVC : BlockTestViewController = BlockTestViewController()
        self.navigationController?.pushViewController(blockVC, animated: true)
    }
    
    final func pushLoopVC() {
        let blockVC : BlockLoopViewController = BlockLoopViewController.init()
        self.navigationController?.pushViewController(blockVC, animated: true)
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

